### PR TITLE
CRUD addresses. Trello Ticket #44

### PR DESCRIPTION
**Pull request by EtriKodoku**
Added functionality for showing list of cities, streets and buildings.

Link to Trello: https://trello.com/c/4IUYPVDe

### Prerequisites:
Log in as admin or staff user

## Features:
1) URL localhost/parser shows a list of cities

2) URL localhost/parser/<city> shows a list of streets

3) URL localhost/parser/<city>/<street> shows a list of buildings

At pages 1, 2 you can find a buttons for deleting records 

## To run server
1. Checkout to directory blackout
```
cd blackout
```
2. Run server
```
python3 manage.py runserver
```

### CODE REVIEW

File: blackout/parserapp/views.py [hunk 1]
```diff
@@ -1,12 +1,52 @@
...
+@staff_member_required()
+def city_view(request, city):
+    streets = Streets.objects.filter(City=city).order_by('Name')
+    context = {'city': city,
+               'streets': streets}
+    return render(request, 'parserapp/streets.html', context=context)
+
+
+@staff_member_required()
+def predelete_view(request, **kwargs):
```
Comment (yurak) [inline blackout/parserapp/views.py:22]: delete_confirmation

File: blackout/parserapp/templates/parserapp/delete/predelete.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Title</title>
+	{%load static%} 
+	<link href="{% static 'parserapp/output.css'%}" rel="stylesheet"> 
+</head>
+<body class="bg-gray-900 text-gray-100/75 pl-2">
+    {% if street == None %}
+        <h2 class="text-2xl">Ви підтверджуєте, що збираєтесь <span class="text-red-500/75">назавжди видалити</span> місто {{city}}?</h2>
```
Comment (yurak) [inline blackout/parserapp/templates/parserapp/delete/predelete.html:11]: use english terms

File: blackout/parserapp/templates/parserapp/delete/predelete.html [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Title</title>
+	{%load static%} 
+	<link href="{% static 'parserapp/output.css'%}" rel="stylesheet"> 
+</head>
+<body class="bg-gray-900 text-gray-100/75 pl-2">
```
Comment (yurak) [inline blackout/parserapp/templates/parserapp/delete/predelete.html:9]: move to css

File: blackout/parserapp/views.py [hunk 4]
```diff
@@ -1,12 +1,52 @@
...
+def delete_view(request, **kwargs):
+    try:
+        Streets.objects.get(City=kwargs['city'], Name=kwargs['street']).delete()  # Delete street
+    except:
+        Streets.objects.filter(City=kwargs['city']).delete()  # Delete city if kwargs['city'] is not defined
+    return redirect('/parser/')
+
+
+@staff_member_required()
+def streets_view(request, street, city):
```
Comment (yurak) [inline blackout/parserapp/views.py:46]: https://studygyaan.com/django/django-crud-create-retrieve-update-delete-function-based-views

File: blackout/parserapp/templates/parserapp/delete/predelete.html (hunk 3 above)
Comment (EtriKodoku) [reply blackout/parserapp/templates/parserapp/delete/predelete.html:9]: It's class names. This names link to CSS

File: blackout/parserapp/views.py [hunk 5]
```diff
@@ -1,12 +1,51 @@
...
+def delete_confirmation(request, **kwargs):
+    """
+    Method for confirming deleting objects. Kwargs contains city and street
+    """
+    try:
+        street = kwargs['street']
+    except:
+        street = None
+    context = {'city': kwargs['city'],
+               'street': street}
```
Comment (Prohodchikoff) [inline blackout/parserapp/views.py:31]: `delete_confirmation(request, city, street = None):` using this code you don't need to use kwargs and try except

File: blackout/parserapp/urls.py [hunk 6]
```diff
@@ -1,4 +1,12 @@
...
 
-urlpatterns = [path("", start, name="start")]
+urlpatterns = [
+    path("", fetch_cities, name="cities"),
+    path("<city>/", fetch_streets, name="city"),
+    path("confirm/<city>/", delete_confirmation, name="delete-confirm-city"),
+    path("confirm/<city>/<street>/", delete_confirmation, name="delete-confirm-street"),
+    path("delete/<city>/", delete_object, name="delete-city"),
+    path("delete/<city>/<street>/", delete_object, name="delete-street"),
+    path("<city>/<street>/", fetch_buildings, name="street")
```
Comment (Prohodchikoff) [inline blackout/parserapp/urls.py:11]: Maybe need add slug field

File: blackout/parserapp/templates/parserapp/streets_list.html [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+{%block title%}Streets List{%endblock%}
+
+{%block content%}
+<div class="pl-2">
+    <h1 class="text-7xl border-b pb-1 mb-4">{{city}}</h1>
+    <a href="{%url 'cities'%}"><button class="p-2 rounded border mr-1">← Назад</button></a>
+    <a href="{%url 'delete-confirm-city' city=city%}"><button class="p-2 rounded border border-red-600 mb-6">Видалити місто</button></a>
+    <ul class="grid">
+        <li class="text-2xl">Вулиці:</li>
```
Comment (Prohodchikoff) [inline blackout/parserapp/templates/parserapp/streets_list.html:11]: Translate the text here and in other places

**Review by Prohodchikoff — CHANGES_REQUESTED** [review]
Check views

**Review by WebUraXalys — APPROVED** [review]

**Review by Prohodchikoff — APPROVED** [review]

**Pull request merged**